osu.Game/Rulesets/Replays/AutoGenerator.cs

```diff
@@ -44,4 +40,37 @@ protected virtual HitObject GetNextObject(int currentIndex)
             return Beatmap.HitObjects[currentIndex + 1];
         }
     }
+
+    public abstract class AutoGenerator<TFrame> : AutoGenerator
+        where TFrame : ReplayFrame
+    {
+        /// <summary>
+        /// The replay frames of the autoplay.
+        /// </summary>
+        protected readonly List<TFrame> Frames = new List<TFrame>();
```

Comment:
Is there a reason we are keeping this as a field, rather than having a `return` value in `GenerateFrames`? Not a huge issue either way, but feels like a return type may be easier to follow.

Reply:
I left it a field so existing code can be adopted to `AutoGenerator<T>` with minimal change.

Reply:
Fair enough. We can probably look at making that (breaking) change in a future iteration. Just feels a bit weird to have that field when its only real use is population and return.